core/src/main/java/com/opensymphony/xwork2/security/DefaultExcludedPatternsChecker.java

```diff
@@ -84,6 +85,14 @@ protected void setDynamicMethodInvocation(String dmiValue) {
         if (!BooleanUtils.toBoolean(dmiValue)) {
             LOG.debug("DMI is disabled, adding DMI related excluded patterns");
             setAdditionalExcludePatterns("^(action|method):.*");
+        } else {
+            LOG.debug("DMI is enabled, adding DMI related ignored patterns");
+            ignoredPatterns = new HashSet<>();
+            try {
+                ignoredPatterns.add(Pattern.compile("^(action|method):.*", Pattern.CASE_INSENSITIVE));
+            } finally {
+            	ignoredPatterns = Collections.unmodifiableSet(ignoredPatterns);
+            }
```

Comment:
Basically right now it doesn't matter if DMI is enabled or not, `action|method` will be excluded. Wouldn't be better to just extend the base RegEx? Or setup this pattern in constructor?

```java
public DefaultExcludedPatternsChecker() {
    setExcludedPatterns(EXCLUDED_PATTERNS);
    setAdditionalExcludePatterns("^(action|method):.*");
}
```

So then additional `ignoredPatterns` field is not needed.